api/core/agent/base_agent_runner.py

```diff
@@ -1,6 +1,7 @@
 import json
 import logging
 import uuid
+from collections.abc import Mapping, Sequence
```

Comment:
BTW, `Mapping` and `Sequence` are abstract types but not typing , which is related to implementation.

Reply:
Thanks for the reminder! Using them here is to remind everyone that this container is immutable.

Reply:
Sure. It's not a blocker issue.